### ISSUE TITLE
Update MODS schema

### ISSUE THREAD

**Issue by alanorth** [DSpace/DSpace#8885]
**Is your feature request related to a problem? Please describe.**
I am not very familiar with MODS. We recently had a request to improve the MODS crosswalk in our repository in order for us to be harvested more thoroughly and I noticed that DSpace 6.4/7.6 currently uses MODS version 3.1, which was [released in 2005](https://www.loc.gov/standards/mods/changes-3-1.html).

**Describe the solution you'd like**
Update the MODS schema to a more modern version. From what I understand the newer versions are much more expressive.

**Describe alternatives or workarounds you've considered**

**Additional context**
The MODS website shows that [version 3.8 has been released](https://www.loc.gov/standards/mods/changes-3-8.html) on 2022-05-22.

**Comment by tdonohue** [comment]
Agreed, but the priority is unclear to me.  I'm not sure how much the MODS metadata format is used via OAI-PMH (as that's the only place DSpace uses MODS).  For now, I'm flagging as "help wanted" but leaving it unscheduled.  I'm glad to pull this over to a board though if we find a volunteer who wants to work on this.

**Comment by alanorth** [comment]
Thanks @tdonohue. Agree the priority is very low and needs a volunteer to take it forward (will probably be me). We had a request by FAO for us to improve the MODs mappings in our repository and I noticed that the DSpace implementation essentially hasn't been touched in twenty years. I wanted to get an issue filed to at least get it on the record here.
